kasa/smartdevice.py

```diff
@@ -379,7 +386,7 @@ def update_from_discover_info(self, info: Dict[str, Any]) -> None:
 
     def _set_sys_info(self, sys_info: Dict[str, Any]) -> None:
         """Set sys_info."""
-        self._sys_info = sys_info
+        self._sys_info = sys_info if "model" in sys_info else sys_info["system"]
```

Comment:
Add a quick command why this is done.

Reply:
I saw you did check earlier if the key `system` is inside the sysinfo, this probably should do the same for consistency?

Reply:
This was forgotten :-)